src/controllers/blog-food.ts

```diff
@@ -0,0 +1,49 @@
+import { UUID } from "crypto";
+import { ErrorCode } from "../exceptions/root";
+import { UnprocessableEntityException } from "../exceptions/validation";
+import { BlogFoodRepository } from "../infrastructure/blog-food-impl";
+import { BlogFoodSchema } from "../schemas/blog-foods";
+import { BlogFoodService } from "../services/blog-food";
+import { Request, Response } from "express";
+import { IdSchema } from "../schemas/id";
+
+const blogFoodRepository = new BlogFoodRepository();
+const blogFoodService = new BlogFoodService(blogFoodRepository);
+
+export const createBlogFood = async (req: Request, res: Response) => {
+    const parsed = BlogFoodSchema.safeParse(req.body);
+    if (!parsed.success) {
+        throw new UnprocessableEntityException(parsed.error, "Validation error!", ErrorCode.UNPROCESSABLE_ENTITY);
+    }
+
+    const { blog_id, food_id } = parsed.data;
+    const result = await blogFoodService.createBlogfood(blog_id as UUID, food_id as UUID);
+
+   res.status(201).json(result);
+}
+
+export const deleteBlogFood = async (req: Request, res: Response) => {
+    const blogId = IdSchema.safeParse(req.params.blogId);
+    const foodId = IdSchema.safeParse(req.params.foodId);
+    if (!blogId.success) {
+        throw new UnprocessableEntityException(blogId.error, "Invalid blog ID", ErrorCode.INVALID_BLOG_ID);
+    }
+    if (!foodId.success) {
+        throw new UnprocessableEntityException(foodId.error, "Invalid food ID", ErrorCode.INVALID_FOOD_ID);
+    }
+
+    const deletedCount = await blogFoodService.deleteBlogFood(blogId.data as UUID, foodId.data as UUID);
+
+    res.status(204).json({ deletedCount });
+};
+
+export const getFoodsForBlog = async (req: Request, res: Response) => {
+    const parsedId = IdSchema.safeParse(req.params.id);
```

Comment:
these kind of things not needed to verify by zod. It can be easily define inside Request. Let me give an example


```suggestion

// Path param type
export interface UserParams {
  id: string;
}

// Query param type
export interface UserQuery {
  role?: string;
  active?: string; // boolean comes as string in query
}

// Body type
export interface UserBody {
  name: string;
  email: string;
}


```

`post('/:id', (req: Request<UserParams, any, UserBody, UserQuery>, res: Response)`

Reply:
ok bhai